Classes/Akamai/Client.php

```diff
@@ -0,0 +1,259 @@
+
+        if ($recursive === true) {
+            $files = [];
+            /** @var File $file */
+            foreach ($directoryListing->files as $file) {
+                if ($file->isDirectory()) {
+                    $recursiveFiles = $this->dir($file->path, true);
+                    if ($recursiveFiles === null || $recursiveFiles->files === []) {
+                        continue;
+                    }
+                    $files[] = $file->withChildren($recursiveFiles->files);
+                } else {
+                    $files[] = $file;
+                }
+            }
+            $directoryListing = DirectoryListing::fromFiles($path, $files);
+        }
+
+        return $directoryListing;
+    }
+
+    public function delete(Path $path): bool
+    {
+        $this->initialize();
+
+        try {
+            /** @phpstan-ignore-next-line */
+            $this->httpClient->put($this->buildUriPathFromFilePath($path), [
+                'headers' => [
+                    'X-Akamai-ACS-Action' => Action::fromString('delete')->acsActionHeader()
+                ]
+            ]);
+
+            return true;
+        } catch (\Exception $exception) {
+            return false;
+        }
+    }
+
+    public function rmdir(Path $path): bool
+    {
+        $this->initialize();
+
+        try {
+            /** @phpstan-ignore-next-line */
+            $this->httpClient->put($this->buildUriPathFromFilePath($path), [
+                'headers' => [
+                    'X-Akamai-ACS-Action' => Action::fromString('rmdir')->acsActionHeader()
+                ]
+            ]);
+
+            return true;
+        } catch (\Exception $exception) {
+            return false;
+        }
+    }
+
+    /**
+     * @return resource
+     */
+    public function stream(Path $path)
+    {
+        /** @phpstan-ignore-next-line */
+        $response = $this->httpClient->get($this->buildUriPathFromFilePath($path), [
+            'headers' => [
+                'X-Akamai-ACS-Action' => Action::fromString('download')->acsActionHeader(),
+            ]
+        ]);
+
+        $stream = StreamWrapper::getResource($response->getBody());
+        fseek($stream, 0);
+
+        return $stream;
+    }
+
+    public function getFullPath(): Path
+    {
+        return Path::fromString((string) $this->cpCode)
+            ->append(Path::fromString((string) $this->restrictedDirectory))
+            ->append($this->workingDirectory ?? Path::fromString(''));
+    }
+
+    protected function buildUriPathFromFilePath(Path $path): string
+    {
+        return urlencode($this->applyPathPrefix($path));
+    }
+
+    protected function applyPathPrefix(Path $path): Path
+    {
+        return Path::fromString((string) $this->cpCode)
+            ->append(Path::fromString((string) $this->restrictedDirectory))
+            ->append($this->workingDirectory ? $path->append($this->workingDirectory) : $path);
```

Comment:
This will append the working directory after the path ... should be `$path->prepend($this->workingDirectory)`